neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,498 @@
+
+                else:
+                    dt = None
+
+                block_info[anno_key] = dt
+
+        bl_ann.update(block_info)
+        for seg_index in range(1):
+            seg_ann = bl_ann['segments'][seg_index]
+
+            # some attributes don't apply to neo spike channels as these cover only a subpart of
+            # the data of a PL2 spike channel
+            spike_annotation_keys = ['m_SortEnabled', 'm_SortRangeStart', 'm_SortRangeEnd',
+                               'm_SourceTrodality', 'm_OneBasedTrode', 'm_OneBasedChannelInTrode',
+                               'm_Source', 'm_Channel']
+            for spike_channel_idx, spike_header in enumerate(self.header['spike_channels']):
+                schannel_name = spike_header['name'].split('.')[0]
+                schannel_info = self.pl2reader.pl2_get_spike_channel_info_by_name(schannel_name)
+
+                spiketrain_an = seg_ann['spikes'][spike_channel_idx]
+                for key in spike_annotation_keys:
+                    spiketrain_an[key] = getattr(schannel_info, key)
+
+
+            event_annotation_keys = ['m_Source', 'm_Channel', 'm_Name']
+            for event_channel_idx, event_header in enumerate(self.header['event_channels']):
+                dchannel_name = event_header['name']
+                dchannel_info = self.pl2reader.pl2_get_digital_channel_info_by_name(dchannel_name)
+
+                event_an = seg_ann['events'][event_channel_idx]
+                for key in event_annotation_keys:
+                    event_an[key] = getattr(dchannel_info, key)
+
+            signal_array_annotation_keys = [
+                'm_SourceTrodality', 'm_OneBasedTrode', 'm_OneBasedChannelInTrode',
+                'm_Source', 'm_Channel', 'm_Name', 'm_MaximumNumberOfFragments'
+            ]
+
+            for stream_idx, stream_header in enumerate(self.header['signal_streams']):
+                signal_array_annotations = {key: [] for key in signal_array_annotation_keys}
+                stream_id = stream_header['id']
+                # extract values of individual signals
+
+                stream_channel_mask = self.header['signal_channels']['stream_id'] == stream_id
+                for signal_idx, signal_header in enumerate(self.header['signal_channels'][stream_channel_mask]):
+                    achannel_name = signal_header['name']
+                    achannel_info = self.pl2reader.pl2_get_analog_channel_info_by_name(achannel_name)
+                    for key in signal_array_annotation_keys:
+                        signal_array_annotations[key].append(getattr(achannel_info, key))
+
+                seg_ann['signals'][stream_idx]['__array_annotations__'] = signal_array_annotations
+
+
+    def _segment_t_start(self, block_index, seg_index):
+        # this must return an float scale in second
+        # this t_start will be shared by all object in the segment except AnalogSignal
```

Comment:
This comments is from the example no ?

Reply:
Yes, i kept some of the comments from the example io, but this one is not applicable any more. I removed it.